impl/src/main/java/io/kroxylicious/testing/kafka/common/Utils.java

```diff
@@ -32,4 +49,30 @@ public static Stream<Integer> preAllocateListeningPorts(int num) {
             throw new UncheckedIOException(e);
         }
     }
+
+    public static void ensureExpectedBrokerCountInCluster(Map<String, Object> connectionConfig, int timeout, TimeUnit timeUnit, Integer expectedBrokerCount) {
```

Comment:
I wonder if `awaitExpectedBrokerCountInCluster` would be better, since it conveys the fact that this method blocks. It's also worth javadocing the known limitation: That the Admin hits a broker which has the metadata with the expected broker count, but that later requests are sent to another broker which hasn't seen that metadata. I.e. this method provides a best-effort check, rather than a true guarantee.

Reply:
Since the aim is test reliability here, would it be better to connect the admin client direct to each broker and proceed only once all brokers have given the correct result?

Reply:
`awaitExpectedBrokerCountInCluster` works as a name probably slightly better as `ensure` could be read to imply it will re-configure the cluster.

The debate when I first raised this was waiting for metadata consistency was hiding the distributed nature of Kafka and would slow down the majority of tests compared to the few that really care about metadata consistency. I'm comfortable with us verifying that a broker is reporting the correct count as good enough.

Reply:
I have merged, we can update it to check each broker in turn via a separate PR if there is a agreement.